### PR TITLE
feat: encryption keys location can be specified by the storage implementation

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
This allows storage implementations to specify the location for encryption keys
https://github.com/owncloud/files_spaces needs this because the space does not belong to a user.

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/4687

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by cdamken** [comment]
Is there a test from the keys that are already in the users to be moved in the spaces?

**Comment by DeepDiver1975** [comment]
> Is there a process to move the keys that are already on the users to be moved in the spaces?

no - and I doubt that there is a general applicable solution.

keys are stored in the individual user's encryption key location.
I'd perform a manual analysis of the explicit system and move key files as needed ...

File: lib/private/Encryption/Util.php [hunk 1]
```diff
@@ -104,6 +107,9 @@ public function __construct(
 		$this->excludedPaths[] = 'avatars';
 		$this->excludedPaths[] = 'avatar.png';
 		$this->excludedPaths[] = 'avatar.jpg';
+
+		// TODO: find a sophisticated solution
```
Comment (DeepDiver1975) [inline lib/private/Encryption/Util.php:111]: Yeah - that's fishy ....

Comment (JammingBen) [reply lib/private/Encryption/Util.php:111]: I'm afraid someone else has to come up with a more sophisticated solution, I don't see a general approach without mentioning the spaces app or providing a public interface to hook up to. This goes not only for this line here - we have multiple references to the spaces app, as you probably already have seen :(

**Pull request closed without merging**